core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -39,68 +37,144 @@ import org.apache.spark.util.Utils
  *
  * So, when adding new fields, take into consideration that the whole object can be serialized for
  * shipping off at any time to consumers of the SparkListener interface.
+ *
+ * @param initialAccums the initial set of accumulators that this [[TaskMetrics]] depends on.
+ *                      Each accumulator in this initial set must be uniquely named and marked
+ *                      as internal. Additional accumulators registered later need not satisfy
```

Comment:
Ah, I see that you already have these checks when constructing the map further down in the constructor.